x-pack/test/cases_api_integration/security_and_spaces/tests/common/internal/bulk_delete_file_attachments.ts

```diff
@@ -0,0 +1,790 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { CaseResponse } from '@kbn/cases-plugin/common';
+import { constructFileKindIdByOwner } from '@kbn/cases-plugin/common/files';
+import { Owner } from '@kbn/cases-plugin/common/constants/types';
+import { getFilesAttachmentReq, getPostCaseRequest } from '../../../../common/lib/mock';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import {
+  bulkCreateAttachments,
+  bulkGetAttachments,
+  createAndUploadFile,
+  createCase,
+  createFile,
+  deleteAllCaseItems,
+  deleteAllFiles,
+  bulkDeleteFileAttachments,
+  getComment,
+  listFiles,
+} from '../../../../common/lib/api';
+import { SECURITY_SOLUTION_FILE_KIND } from '../../../../common/lib/constants';
+import {
+  globalRead,
+  noKibanaPrivileges,
+  superUser,
+} from '../../../../common/lib/authentication/users';
+import { createUsersAndRoles, deleteUsersAndRoles } from '../../../../common/lib/authentication';
+import {
+  casesAllUser,
+  obsCasesAllUser,
+  obsCasesReadUser,
+  obsSecCasesAllUser,
+  obsSecCasesReadUser,
+  secAllSpace1User,
+  secAllUser,
+  secReadUser,
+  users as api_int_users,
+} from '../../../../../api_integration/apis/cases/common/users';
+import { roles as api_int_roles } from '../../../../../api_integration/apis/cases/common/roles';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const es = getService('es');
+
+  describe('bulk_delete_file_attachments', () => {
+    // we need api_int_users and roles because they have authorization for the actual plugins (not the fixtures). This
+    // is needed because the fixture plugins are not registered as file kinds
+    before(async () => {
+      await createUsersAndRoles(getService, api_int_users, api_int_roles);
+    });
+
+    after(async () => {
+      await deleteUsersAndRoles(getService, api_int_users, api_int_roles);
+    });
+
+    describe('failures', () => {
+      let postedCase: CaseResponse;
+
+      before(async () => {
+        postedCase = await createCase(supertest, getPostCaseRequest());
+      });
+
+      after(async () => {
+        await deleteAllFiles({
+          supertest,
+        });
+        await deleteAllCaseItems(es);
+      });
+
+      it('fails to delete a file when the file does not exist', async () => {
+        await bulkDeleteFileAttachments({
+          supertest,
+          caseId: postedCase.id,
+          fileIds: ['abc'],
+          expectedHttpCode: 500,
```

Comment:
Great point. The issue here was that the file service isn't returning the saved object error (which would have been a boom) it was wrapping it in its own `FileNotFound` error class. Our cases code only looks for booms to grab the status from. So since our code was encountering a non-boom it defaults to 500. I added a simple fix to do an `error instanceof FileNotFound` so we can get the 400. The file service does a similar check before it returns from their route.